### PR TITLE
core/types: dependency cleanup

### PR DESCRIPTION
**Pull request by fjl**
This PR shrinks the core/types dependency closure. It previously depended on core/vm, p2p/discover, p2p/discv5 and go-leveldb. The main motivation is to reduce the amount of code vendored by go-ethereum library users who only need ethclient.

This checks almost all boxes for resolving https://github.com/ethereum/go-ethereum/issues/3289, the only thing left to do would be to move the crypto out of core/types.

### CODE REVIEW

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -485,17 +485,15 @@ func makeNodeUserIdent(ctx *cli.Context) string {
...
+	if ctx.GlobalIsSet(BootnodesFlag.Name) {
+		urls = ctx.GlobalStringSlice(BootnodesFlag.Name)
+	} else if ctx.GlobalBool(TestNetFlag.Name) {
+		urls = params.TestnetBootnodes
 	}
-	// Otherwise parse and use the CLI bootstrap nodes
-	bootnodes := []*discover.Node{}
 
-	for _, url := range strings.Split(ctx.GlobalString(BootnodesFlag.Name), ",") {
+	var bootnodes []*discover.Node
```
Comment (karalabe) [inline cmd/utils/flags.go]: How about `bootnodes := make([]*discover.Node, 0, len(urls))` ?

File: cmd/utils/flags.go [hunk 2]
```diff
@@ -509,14 +507,13 @@ func MakeBootstrapNodes(ctx *cli.Context) []*discover.Node {
...
-		return params.DiscoveryV5Bootnodes
+	urls := params.DiscoveryV5Bootnodes
+	if ctx.GlobalIsSet(BootnodesFlag.Name) {
+		urls = ctx.GlobalStringSlice(BootnodesFlag.Name)
 	}
-	// Otherwise parse and use the CLI bootstrap nodes
-	bootnodes := []*discv5.Node{}
 
-	for _, url := range strings.Split(ctx.GlobalString(BootnodesFlag.Name), ",") {
+	var bootnodes []*discv5.Node
```
Comment (karalabe) [inline cmd/utils/flags.go]: How about `bootnodes := make([]*discv5.Node, 0, len(urls))` ?

**Comment by karalabe** [comment]
```
I0106 14:34:10.183344 eth/sync.go:190] fast sync complete, auto disabling
I0106 14:34:16.888935 core/blockchain.go:1073] imported 2 blocks,    28 txs (  0.747 Mg) in  51.853ms (14.405 Mg/s). #2946045 [556340e6… / bfdfee0c…]
I0106 14:34:18.179474 core/blockchain.go:1073] imported 1 blocks,     3 txs (  0.063 Mg) in   6.644ms ( 9.482 Mg/s). #2946046 [cbc64c29…]
```

LGTM

**Comment by karalabe** [comment]
Please tidy up the commits count though (squash, at least partially the fixes).

**Comment by fjl** [comment]
squashed

**Pull request merged**